docs/source/conf.py

```diff
@@ -30,18 +30,21 @@
 # https://www.sphinx-doc.org/en/master/usage/configuration.html#general-configuration
 
 extensions = [
-    "sphinx.ext.napoleon",
-    "sphinx.ext.autodoc",
-    "sphinx.ext.autosectionlabel",
-    "sphinx_autodoc_typehints",
-    "numpydoc",
-    "sphinx_tabs.tabs",
+    "sphinx.ext.napoleon",  # Support parsing numpydoc style docstrings
+    "sphinx.ext.autodoc",  # Support generating API reference from docstrings
+    "sphinx.ext.autosectionlabel",  # Support referencing sections using their titles
+    "sphinx.ext.intersphinx",  # Support referencing external APIs
```

Comment:
I added this `sphinx.ext.intersphinx` extension (built-in to Sphinx) to reference the Python official documentation in certain places (e.g. when I mention `NamedTuple` and `dataclasses`).